sunpy/net/dataretriever/sources/vsm.py

```diff
@@ -0,0 +1,163 @@
+            tmp_pattern = 'k4{obstype}%y%m%dt%H%M%S{suffix}'
+            result = list()
+            base_url = 'http://gong2.nso.edu/pubkeep/{obstype}/{date:%Y%m}/k4{obstype}{date:%y%m%d}/'
+            total_days = (timerange.end - timerange.start).days + 1
+            all_dates = timerange.split(total_days)
+            for day in all_dates:
+                try:
+                    html = urllib2.urlopen(base_url.format(obstype=obstype, date=day.end))
+                    soup = BeautifulSoup(html)
+                    for link in soup.findAll("a"):
+                        crawler = Scraper(tmp_pattern, obstype=obstype, suffix=suffix)
+                        simple_pattern = '%y%m%d_%H%M%S'
+                        url = str(link.get('href'))
+                        if crawler._URL_followsPattern(url):
+                            tmp = re.sub('k4'+obstype, '', url)
+                            tmp = re.sub('[a-zA-Z]', '', tmp)
+                            crawler_ = Scraper(simple_pattern)
+                            if timerange.start <= crawler_._extractDateURL(tmp) <= timerange.end:
+                                result.append(base_url.format(obstype=obstype, date=day.end)+url)
+                except:
+                    pass
+            return result
+
+        start_date = {6302: datetime.datetime(2003, 8, 21), 8542: datetime.datetime(2003, 8, 26),
+                      10830: datetime.datetime(2004, 11, 4)}
+        START_DATE = start_date[wave]
+
+        if timerange.start < START_DATE:
+            raise ValueError('Earliest date for which SOLIS VSM data is available is {:%Y-%m-%d}'.format(START_DATE))
+
+        result = list()
+        if wave == 6302:
+            if not physobs_in:
+                result.extend(download_from_nso('v72', '.fts.gz', timerange))
+                result.extend(download_from_nso('v93', '_FDISK.fts.gz', timerange))
+            else:
+                if kwargs['physobs'] == 'VECTOR_MAGNETIC_FIELD':
+                    result.extend(download_from_nso('v93', '_FDISK.fts.gz', timerange))
+                else:
+                    result.extend(download_from_nso('v72', '.fts.gz', timerange))
+        else:
+            result.extend(download_from_nso(table_wave[wave]), '.fts.gz', timerange)
+        return result
+
+    def _makeimap(self):
+        """
+        Helper Function:used to hold information about source.
+        """
+        self.map_['source'] = 'SOLIS'
+        self.map_['instrument'] = 'vsm'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+
+        """
+        Answers whether client can service the query.
+
+        Parameters
+        ----------
+        query : list of query objects
+
+        Returns
+        -------
+        boolean: answer as to whether client can service the query
+
+        """
+        chkattr = ['Time', 'Instrument', 'Wavelength']
+        chklist = [x.__class__.__name__ in chkattr for x in query]
```

Comment:
Yeah, the current ```Fido``` clients are very different from the previous ones. I was thinking of removing both of them, but they do serve a purpose of telling which attributes the ```Fido``` client takes as input.